src/fcntl.rs

```diff
@@ -305,6 +308,43 @@ pub fn flock(fd: RawFd, arg: FlockArg) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+bitflags! {
+    pub struct CopyFileRangeFlags: c_int {
+        const NONE = 0 as c_int;
+    }
+}
+
+/// Copy a range of data from one file to another
+///
+/// The `copy_file_range` system call performs an in-kernel copy between
+/// file descriptor `fd_in` and `fd_out` without the additional cost of transferring data
+/// from the kernel to user space and then back into the kernel.  It
+/// copies up to `len` bytes of data from file descriptor `fd_in` to file
+/// descriptor `fd_out`, overwriting any data that exists within the
+/// requested range of the target file.
+///
+/// The flags argument is currently unused, and is provided to allow for future
+/// extensions and currently must be set to `CopyFileRangeFlags::empty()`.
+///
+/// On completion the number of bytes actually copied will be returned.
+#[cfg(any(target_os = "android", target_os = "linux"))]
+pub fn copy_file_range(
+    fd_in: RawFd, off_in: Option<&mut libc::loff_t>,
```

Comment:
Why are the offsets provided as mutable references?  If this operation changes them, could you describe how in the doc comment?

Also, it's ugly to expose raw libc types in our public API.  In this case, I think it would be preferable for the arguments to be `&mut i64`.

Reply:
Done

Reply:
I agree.  Returning a struct would be more Rusty than using mutable arguments.

Reply:
Hmm, it would be a bit of an overkill and possibly also a memory management issue to return new structs. Purpose of this method is to quickly copy large parts of files incrementally, so it will most likely be called in an iterative fashion with feedback. Agree to leave it the way it is now, or do we want some extra helper methods?

Reply:
I suppose it would be ok to leave as is.  Returning a structure with offsets could be up to 256 bits.